### ISSUE TITLE
Some (single-line) links are not clickable via hints

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#292]
* On http://www.heise.de/ the two links under the big picture
* On http://geizhals.de/?o=9 the titles
* Titles on http://www.golem.de/
* Member function links on cppreference.com, e.g. `(constructor)` [for std::vector](http://en.cppreference.com/w/cpp/container/vector)

**Comment by boblehest** [comment]
On facebook event pages, you have the following three links on the right:
http://i.imgur.com/aeqbgkI.png
Clicking the "went" (or "going", if the event is in the future) link has no effect, but strangely the other two links work.

**Comment by adskankster** [comment]
I was coming to report this issue, but as it's already here, I'll just post the url for the (main) site where I've found it. The "read more" link on the right hand side here: http://earthobservatory.nasa.gov/
Thanks.

**Comment by The-Compiler** [comment]
@adskankster both the one on the right of the image and the one in the black "features" box work fine here.

**Comment by adskankster** [comment]
There's two to the right of the image, the main heading (which works for me) and the "Read More" one which doesn't. Here's my hint settings from conf:

border = 1px solid #E3BE23
opacity = 0.7
mode = letter
chars = asdfghjkl
min-chars = 1
scatter = true
uppercase = false
auto-follow = true
next-regexes = \bnext\b,\bmore\b,\bnewer\b,\b[>→≫]\b,\b(>>|»)\b
prev-regexes = \bprev(ious)?\b,\bback\b,\bolder\b,\b[<←≪]\b,\b(<<|«)\b


I hadn't looked at the "Features" box, and on looking the "read more" does, as do most others. There are however one or two hints that don't work, but that's because I don't think they're valid links (eg the button to move left in the list in "Features" works, but is also hinted when at the left-most item. There is alos an extra hint that does nothing on the right hand side. These are not a big issue, but it can sometimes make it difficult to know which hint to use on sites that have these sorts of boxes.

**Comment by NoSuck** [comment]
I experience this issue at the following website, in the middle, news column at the top.  (A membership might be necessary to view it properly.)

https://conyac.cc/ja/

**Comment by NoSuck** [comment]
I found another, [on the link for <https://github.com/github/dmca/blob/master/2015-07-06-nintendo.md>.](http://gigazine.net/news/20150713-nintendo-game-boy-emulator)

**Comment by The-Compiler** [comment]
You mean the link to the DCMA notice, on the Gigazine page, right?

That seems to work fine for me - are you sure the window is big enough to show the link on one line? If the link is wrapped, that's another (known) issue.

**Comment by NoSuck** [comment]
Interesting.  You are correct.  I was unaware of the line-wrap issue.

... My tiny monitor has served its purpose.

**Comment by antoyo** [comment]
I started to work on this issue.
Everything seems to work as expected with my fixes, except clicking on the hint when I scroll down the page (thought it works on this site: https://doc.rust-lang.org/stable/book/hello-world.html because the `getBoundingClientRect().top` is always 0).
The hint is at the right position, but I cannot activate it on http://www.golem.de, for instance.

**Comment by The-Compiler** [comment]
Nice! I didn't work on the hint issues mainly because I want to have some test cases for them before I start changing things - but maybe that's a good incentive for me to start working on them after getting the current PRs down :wink:

I'm guessing `getBoundingClientRect()` gives you the rectangle on the whole page (i.e. a point outside of the window when you're scrolled down). The old code called `elem.rect_on_view()` to get the correct position.

There's also a `webelem.rect_on_view()` function which takes a `QWebElement` and a `QRect` - maybe just try calling that and using the `QRect` it returns?

**Comment by error800** [comment]
Testing the links:
- On http://www.heise.de/ the two links under the big picture
Which links are you referring to? Maybe they changed the layout?

- On http://geizhals.de/?o=9 the titles
Works here.

- Titles on http://www.golem.de/
Same problem as in #1298, block level elements in <code>&lt;a&gt;</code>

- Member function links on cppreference.com, e.g. (constructor) for std::vector
Very similar to #1298, elements with <code>display: table;</code> inside <code>&lt;a&gt;</code> also do not work.

```html
<a href="http://qutebrowser.org">
  <span style="display: table;">
    link
  </span>
</a>
```

**Comment by The-Compiler** [comment]
All of those except the geizhals one have been fixed by #1311. For the geizhals one I opened #1514.
